tensorflow/python/autograph/converters/break_statements.py

```diff
@@ -115,37 +130,54 @@ def visit_For(self, node):
     # A break in the else clause applies to the containing scope.
     node.orelse = self.visit_block(node.orelse)
 
-    if break_used:
-      # Python's else clause only triggers if the loop exited cleanly (e.g.
-      # break did not trigger).
-      guarded_orelse = self._guard_if_present(node.orelse, break_var)
-      extra_test = templates.replace_as_expression(
-          'ag__.not_(var_name)', var_name=break_var)
-
-      # The extra test is hidden in the AST, which will confuse the static
-      # analysis. To mitigate that, we insert a no-op statement that ensures
-      # the control variable is marked as used.
-      # TODO(mdan): Use a marker instead, e.g. ag__.condition_loop_on(var_name)
+    if not break_used:
       template = """
-        var_name = False
         for target in iter_:
-          (var_name,)
           body
-        else:
-          orelse
+        orelse
       """
       node = templates.replace(
           template,
-          var_name=break_var,
           iter_=node.iter,
           target=node.target,
           body=node.body,
-          orelse=guarded_orelse)
+          orelse=node.orelse)
 
-      new_for_node = node[1]
-      anno.setanno(new_for_node, anno.Basic.EXTRA_LOOP_TEST, extra_test)
+      new_for_node = node[0]
+      anno.copyanno(original_node, new_for_node, 'extra_test')
```

Comment:
Please keep the `anno.Basic.EXTRA_LOOP_TEST` here.

Reply:
changed.